apis/v1beta1/config.go

```diff
@@ -133,6 +139,58 @@ type Config struct {
 	Service    Service    `json:"service" yaml:"service"`
 }
 
+// getPortsForComponentTypes gets the ports for the given ComponentType(s).
+// NOTE FOR REVIEWERS: we could also do this by introducing something in components that is aware of which retriever
+// is being used and pass the right config in where necessary...
+func (c *Config) getPortsForComponentTypes(logger logr.Logger, componentType ...ComponentType) ([]corev1.ServicePort, error) {
```

Comment:
We should try to use the same terminology as the collector here. `receiver` is the component [`kind`](https://github.com/open-telemetry/opentelemetry-collector/blob/227fb823dbf33e6a62b842aef5a70e10ed1db84f/component/component.go#L88), whereas `prometheus` is the component [`type`.](https://github.com/open-telemetry/opentelemetry-collector/blob/227fb823dbf33e6a62b842aef5a70e10ed1db84f/component/config.go#L100)

Reply:
I think i will clear up the type/name stuff in a followup because it's a bit bigger of a change.